test/integration_test.go

```diff
@@ -2515,6 +2517,170 @@ func (ts *IntegrationTestSuite) TestHeartbeatThrottleDisabled() {
 	ts.assertReportedOperationCount("temporal_request_failure", "RecordActivityTaskHeartbeat", 0)
 }
 
+func (ts *IntegrationTestSuite) TestUpsertMemoFromNil() {
+	// TODO(rodrigozhou): unskip after UpsertMemo is released
+	ts.T().Skip("Not implemented in server yet")
```

Comment:
My approval of the PR is pending @bergundy and @yiminc and others coming to a conclusion on whether we want a capability boolean so users can check whether their server version supports this.

Reply:
Updated to check capability.